packages/@aws-cdk/aws-scheduler-alpha/lib/schedule.ts

```diff
@@ -222,9 +263,11 @@ export class Schedule extends Resource implements ISchedule {
       target: {
         arn: targetConfig.arn,
         roleArn: targetConfig.role.roleArn,
-        input: targetConfig.input?.bind(this),
+        input: props.targetOverrides?.input ?
+          props.targetOverrides?.input?.bind(this) :
+          targetConfig.input?.bind(this),
         deadLetterConfig: targetConfig.deadLetterConfig,
-        retryPolicy: targetConfig.retryPolicy,
+        retryPolicy: retryPolicy.maximumEventAgeInSeconds || retryPolicy.maximumRetryAttempts ? retryPolicy : undefined,
```

Comment:
I don't like this, it's unnecessarily complicated. And what if we add a prop to `retryPolicy` in the future? Lets add a function that creates a retryPolicy but returns undefined if no properties inside the object are defined